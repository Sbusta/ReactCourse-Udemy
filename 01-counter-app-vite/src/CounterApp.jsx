import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter,setCaunter] = useState(value)

    const handkeAdd =() => {
        setCaunter(counter + 1);
    }

    const handkeSubtract =() => {
        setCaunter(counter - 1);
    }
    
    const handkeReset =() => {
        setCaunter(value);
    }

    return(
    <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>

        <button onClick={ handkeAdd }> +1 </button>
        <button onClick={ handkeSubtract }> -1 </button>
        <button onClick={ handkeReset }> Reset </button>

    </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}