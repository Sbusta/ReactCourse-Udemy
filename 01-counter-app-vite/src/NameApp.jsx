import PropTypes from 'prop-types';

export const NameApp = ({title, subTitle, name, job}) => {
    return (
        <>
            <h1>{name}</h1>
            <p>{job}</p>
            <h3>{title}</h3>
            <h4>{subTitle}</h4>
        </>
    )
  }

NameApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number,
}

NameApp.defaultProps = {
    
    subTitle: 2516,
    name: 'Sebasti√°n Bustamante',
    job: 'Software Developer',
}