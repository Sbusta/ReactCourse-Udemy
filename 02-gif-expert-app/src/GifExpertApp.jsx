import { useState } from "react"
import { AddCategory,GifGrid } from "./components"

export const GifExpertApp = () => {
  
  const [categories, setCategories] = useState(['Star Wars'])

  const onAddCategory = (newCategory) =>{

    if(categories.includes(newCategory)) return;

    setCategories([newCategory, ...categories])
      
  }
  return (
    <>
        {/* title */}
        <h1>GifExpertApp</h1>

        {/* Input */}
        <AddCategory 
        //setCategories={setCategories} 
          onNewCategory={ onAddCategory }
        />
        {/* Gif list*/} 
        
        {
          categories.map(category =>(
            <GifGrid 
            key={category} 
            category={ category } />
          ))
        }
        
        {/* Gif Item */}
    </>
  )
}
