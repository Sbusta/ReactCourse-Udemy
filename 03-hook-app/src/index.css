html, body{ 
    /* background-color: #21232A; */
    color: #21232A;
    padding: 20px;
}

button{
    margin-right: 10px;
}