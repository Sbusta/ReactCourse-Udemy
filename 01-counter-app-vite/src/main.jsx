import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
//import { HelloWorldApp } from './HelloWorldApp'
//import { NameApp } from './NameApp'
import { CounterApp } from './CounterApp'




ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/*<NameApp title= 'Buen TÃ­tulo' />*/}
    <CounterApp value={20} />
  </React.StrictMode>,
)
