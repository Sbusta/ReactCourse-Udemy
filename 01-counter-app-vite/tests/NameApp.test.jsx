import { render, screen } from "@testing-library/react";
import { NameApp } from "../src/NameApp"


describe('Pruebas de NameApp', () => { 
    const title = 'Buen Título'
    const subtitle = 2516

    test('debe de ser igual al snapshot', () => { 
        const {container} = render(<NameApp title={title}/>);

        expect(container).toMatchSnapshot();

     })
     test('titulo debe de ser igual "Buen Título"', () => { 
        render(<NameApp title={title}/>);

        expect(screen.getByText(title)).toBeTruthy();
        
     })
     test('titulo en un h1', () => { 

        render(<NameApp title={title}/>);

        expect(screen.getByRole('heading',{level:3}).innerHTML).toContain(title);
     })

     test('subtitulo pasado por la sp props', () => { 

        render(<NameApp 
                    title={title}
                    subtitle={subtitle}

                />);

        expect(screen.getAllByText(subtitle).length).toBe(1);
     })
 })