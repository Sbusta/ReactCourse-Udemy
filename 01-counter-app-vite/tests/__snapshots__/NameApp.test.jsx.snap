// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de NameApp debe de ser igual al snapshot 1`] = `
<div>
  <h1>
    Sebastián Bustamante
  </h1>
  <p>
    Software Developer
  </p>
  <h3>
    Buen Título
  </h3>
  <h4>
    2516
  </h4>
</div>
`;
